File: cs50x/1-c/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void loop_and_print(int length, string val);

int main(void)
{
    int num;
    do
    {
        num = get_int("Height: ");
    }
    while (num < 1 || num > 8);

    for (int i = 1; i <= num; i++)
    {
        loop_and_print(num - i, " ");
        loop_and_print(i, "#");
        printf("  ");
        loop_and_print(i, "#");
        printf("\n");
    }
}

void loop_and_print(int length, string val)
{
    for (int i = 0; i < length; i ++)
    {
        printf("%s", val);
    }
}

File: cs50x/6-python/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    while True:
        number = get_int("Height: ")
        if number >= 1 and number <= 8:
            for i in range(number):
                i += 1
                print(" " * (number - i), print_hash(i), "  ", print_hash(i), sep="")
            return


def print_hash(i):
    return "#" * i


main()